File: cash.py
from cs50 import get_float

# Prompt user for their change owed
while True:
    dollars = get_float("Change owed: ")
# Makes sure the user inputs a vaild number
    if dollars >= 0:
        break

# Finds out the total cents
cents = round(dollars * 100)
# initializes coins to 0
coins = 0

# Removes cents and counts coins
while cents >= 25:
    cents -= 25
    coins += 1
    
while cents >= 10:
    cents -= 10
    coins += 1

while cents >= 5:
    cents -= 5
    coins += 1

while cents >= 1:
    cents -= 1
    coins += 1

# Total number of coins owed
print(coins)

File: Week1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h> // Used for rounding function only

int main(void)
{
    float dollars; // Declare dollars
    do
    {
        dollars = get_float("Change owed: "); // Prompt user to input the total change owed and store it in the variable dollars 
    }

    while (dollars < 0); // Error checking user input 

    int cents = round(dollars * 100); // Calculates total cents by multiplying dollars by 100 then rounding off
    int coins = 0; // Declare coins 

    while (cents >= 25) // Increments coins by 1 so long as cents is greater than or equal to 25
    {
        cents -= 25; // Subtract 25 from cents
        coins++; // Increment coins by 1
    }
    while (cents >= 10) // Once cents is less than 25 increment coins by 1
    {
        cents -= 10; // Subtract 10 from cents
        coins++; // Increment coins by 1
    }
    while (cents >= 5) // Once cents is less than 10 increment coins by 1
    {
        cents -= 5; // Subtract 5 from cents
        coins++; // Increment coins by 1
    }
    while (cents >= 1) // If there is 1 cent left over increment coins by 1
    {
        cents -= 1; // Subtract 1 from cents
        coins++; // Increment coins by 1
    }
    printf("%i\n", coins); // Print total number of coins 
}
